### PR TITLE
adding docstrings to emr.py

### PR DESCRIPTION
**Pull request by joshuaroot**
* adding docstrings to emr.py

### CODE REVIEW

File: c7n/resources/emr.py [hunk 1]
```diff
@@ -75,6 +75,23 @@ def augment(self, resources):
...
+    It is recommended to apply a filter to the terminate action to avoid
+    termination of all EMR clusters
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: emr-terminate
+                resource: emr
```
Comment (kapilt) [inline c7n/resources/emr.py]: i think we need to document emr with query filters on state, as the service keeps history of previous job flow clusters.

Comment (kapilt) [inline c7n/resources/emr.py]:
```
  - name: emr-clusters
    description:
        Report on any EMR clusters
    resource: emr
    query:
        - ClusterStates: [STARTING, BOOTSTRAPPING, RUNNING, WAITING]
```

please ensure that examples are tested, i'm not really clear which of these pull requests on doc examples have been fully vetted and which have not. fwiw, the one above has been.

Comment (joshuaroot) [inline c7n/resources/emr.py]: Ok, I'll go through each docstring PR I have and once I've ensured all examples are valid & tested, I'll comment on them letting you know.

Comment (kapilt) [inline c7n/resources/emr.py]: most of the prs are not commented, leaving a dozen prs in wip is not cool, especially while opening more wip prs. i'm closing out the ones that are not obviously correct at this point.

Comment (kapilt) [inline c7n/resources/emr.py]: please reopen the prs when they are complete

**Pull request merged**